### ISSUE TITLE
Type error after magit update

### ISSUE THREAD

**Issue by JimMoen** [alphapapa/magit-todos#183]
```txt
magit         20240721.1509
magit-todos   20240519.247
```
error message:
```elisp
rxt-adt->pcre/lev: Wrong type argument: rxt-seq, #s(rxt-choice (#s(rxt-seq (#s(rxt-primitive "^" bol) #s(rxt-submatch #s(rxt-repeat 1 nil #s(rxt-string "*" nil) t)) #s(rxt-repeat 1 nil #s(rxt-char-set-union nil nil (blank) nil) t) #s(rxt-submatch #s(rxt-choice (#s(rxt-string "DEBUG" nil) #s(rxt-string "FIXME" nil) #s(rxt-string "GOTCHA" nil) #s(rxt-string "STUB" nil) #s(rxt-string "TODO" nil) #s(rxt-string "XXX" nil)))) #s(rxt-repeat 1 nil #s(rxt-char-set-union nil nil (space) nil) t) #s(rxt-submatch #s(rxt-repeat 1 nil #s(rxt-primitive "." nonl) t)))) #s(rxt-seq (#s(rxt-choice (#s(rxt-primitive "^" bol) #s(rxt-repeat 1 nil #s(rxt-char-set-union nil nil (blank) nil) t))) #s(rxt-submatch #s(rxt-choice (#s(rxt-string "DEBUG" nil) #s(rxt-string "FIXME" nil) #s(rxt-string "GOTCHA" nil) #s(rxt-string "STUB" nil) #s(rxt-string "TODO" nil) #s(rxt-string "XXX" nil)))) #s(rxt-repeat 0 1 #s(rxt-seq (#s(rxt-char-set-union (91 40) nil nil nil) #s(rxt-repeat 1 nil #s(rxt-char-set-negation #s(rxt-char-set-union ... nil nil nil)) t) #s(rxt-char-set-union (41 93) nil nil nil))) t) #s(rxt-string ":" nil) #s(rxt-repeat 0 1 #s(rxt-seq (#s(rxt-repeat 1 nil #s(rxt-char-set-union nil nil (blank) nil) t) #s(rxt-submatch #s(rxt-repeat 1 nil #s(rxt-primitive "." nonl) t)))) t)))))Invalid face attribute :foreground nil
```

**Comment by alphapapa** [comment]
Why do you think this is a problem with magit-todos?  You didn't provide the backtrace, so there's not even a way to see what function the error was signaled from.  And if you updated Magit and the error started happening, why would you blame this package instead of Magit?

**Comment by JimMoen** [comment]
Let me describe the problem I encountered in another way:
Fresh instatll `magit` and `magit-todos`.
confs for `magit-todos`:
```elisp
(use-package magit-todos
  :after magit
  :config
  (magit-todos-mode 1)
  (setq magit-todos-keywords-list
        '("XXX" "TODO" "FIXME" "DEBUG" "GOTCHA" "STUB")))
```

The reason I think the error is caused by magit-todos:
The above comment contains several todos keywords I set: "XXX" "TODO" "FIXME" "DEBUG" "GOTCHA" "STUB"

Steps:
- eval: `(setq debug-on-error t)`
- entry a repo with `magit-status`
- see the following error:

<details>
<summary>Debugger entered--Lisp error:</summary> 

```elisp
Debugger entered--Lisp error: (wrong-type-argument rxt-seq #s(rxt-choice :elts (#s(rxt-seq :elts (#s(rxt-primitive :pcre "^" :rx bol) #s(rxt-submatch :body #s(rxt-repeat :from 1 :to nil :body #s(rxt-string :chars "*" :case-fold nil) :greedy t)) #s(rxt-repeat :from 1 :to nil :body #s(rxt-char-set-union :chars nil :ranges nil :classes (blank) :case-fold nil) :greedy t) #s(rxt-submatch :body #s(rxt-choice :elts (... ... ... ... ... ...))) #s(rxt-repeat :from 1 :to nil :body #s(rxt-char-set-union :chars nil :ranges nil :classes (space) :case-fold nil) :greedy t) #s(rxt-submatch :body #s(rxt-repeat :from 1 :to nil :body #s(rxt-primitive :pcre "." :rx nonl) :greedy t)))) #s(rxt-seq :elts (#s(rxt-choice :elts (#s(rxt-primitive :pcre "^" :rx bol) #s(rxt-repeat :from 1 :to nil :body ... :greedy t))) #s(rxt-submatch :body #s(rxt-choice :elts (... ... ... ... ... ...))) #s(rxt-repeat :from 0 :to 1 :body #s(rxt-seq :elts (... ... ...)) :greedy t) #s(rxt-string :chars ":" :case-fold nil) #s(rxt-repeat :from 0 :to 1 :body #s(rxt-seq :elts (... ...)) :greedy t))))))
  rxt-seq->pcre(#s(rxt-choice :elts (#s(rxt-seq :elts (#s(rxt-primitive :pcre "^" :rx bol) #s(rxt-submatch :body #s(rxt-repeat :from 1 :to nil :body #s(rxt-string :chars "*" :case-fold nil) :greedy t)) #s(rxt-repeat :from 1 :to nil :body #s(rxt-char-set-union :chars nil :ranges nil :classes (blank) :case-fold nil) :greedy t) #s(rxt-submatch :body #s(rxt-choice :elts (... ... ... ... ... ...))) #s(rxt-repeat :from 1 :to nil :body #s(rxt-char-set-union :chars nil :ranges nil :classes (space) :case-fold nil) :greedy t) #s(rxt-submatch :body #s(rxt-repeat :from 1 :to nil :body #s(rxt-primitive :pcre "." :rx nonl) :greedy t)))) #s(rxt-seq :elts (#s(rxt-choice :elts (#s(rxt-primitive :pcre "^" :rx bol) #s(rxt-repeat :from 1 :to nil :body ... :greedy t))) #s(rxt-submatch :body #s(rxt-choice :elts (... ... ... ... ... ...))) #s(rxt-repeat :from 0 :to 1 :body #s(rxt-seq :elts (... ... ...)) :greedy t) #s(rxt-string :chars ":" :case-fold nil) #s(rxt-repeat :from 0 :to 1 :body #s(rxt-seq :elts (... ...)) :greedy t))))))
  rxt-adt->pcre/lev(#s(rxt-choice :elts (#s(rxt-seq :elts (#s(rxt-primitive :pcre "^" :rx bol) #s(rxt-submatch :body #s(rxt-repeat :from 1 :to nil :body #s(rxt-string :chars "*" :case-fold nil) :greedy t)) #s(rxt-repeat :from 1 :to nil :body #s(rxt-char-set-union :chars nil :ranges nil :classes (blank) :case-fold nil) :greedy t) #s(rxt-submatch :body #s(rxt-choice :elts (... ... ... ... ... ...))) #s(rxt-repeat :from 1 :to nil :body #s(rxt-char-set-union :chars nil :ranges nil :classes (space) :case-fold nil) :greedy t) #s(rxt-submatch :body #s(rxt-repeat :from 1 :to nil :body #s(rxt-primitive :pcre "." :rx nonl) :greedy t)))) #s(rxt-seq :elts (#s(rxt-choice :elts (#s(rxt-primitive :pcre "^" :rx bol) #s(rxt-repeat :from 1 :to nil :body ... :greedy t))) #s(rxt-submatch :body #s(rxt-choice :elts (... ... ... ... ... ...))) #s(rxt-repeat :from 0 :to 1 :body #s(rxt-seq :elts (... ... ...)) :greedy t) #s(rxt-string :chars ":" :case-fold nil) #s(rxt-repeat :from 0 :to 1 :body #s(rxt-seq :elts (... ...)) :greedy t))))))
  rxt-adt->pcre(#s(rxt-choice :elts (#s(rxt-seq :elts (#s(rxt-primitive :pcre "^" :rx bol) #s(rxt-submatch :body #s(rxt-repeat :from 1 :to nil :body #s(rxt-string :chars "*" :case-fold nil) :greedy t)) #s(rxt-repeat :from 1 :to nil :body #s(rxt-char-set-union :chars nil :ranges nil :classes (blank) :case-fold nil) :greedy t) #s(rxt-submatch :body #s(rxt-choice :elts (... ... ... ... ... ...))) #s(rxt-repeat :from 1 :to nil :body #s(rxt-char-set-union :chars nil :ranges nil :classes (space) :case-fold nil) :greedy t) #s(rxt-submatch :body #s(rxt-repeat :from 1 :to nil :body #s(rxt-primitive :pcre "." :rx nonl) :greedy t)))) #s(rxt-seq :elts (#s(rxt-choice :elts (#s(rxt-primitive :pcre "^" :rx bol) #s(rxt-repeat :from 1 :to nil :body ... :greedy t))) #s(rxt-submatch :body #s(rxt-choice :elts (... ... ... ... ... ...))) #s(rxt-repeat :from 0 :to 1 :body #s(rxt-seq :elts (... ... ...)) :greedy t) #s(rxt-string :chars ":" :case-fold nil) #s(rxt-repeat :from 0 :to 1 :body #s(rxt-seq :elts (... ...)) :greedy t))))))
  rxt-elisp-to-pcre("\\(?:^\\(\\*+\\)[[:blank:]]+\\(\\(?:DEBUG\\|FIXME\\|GOTCHA\\|STUB\\|TODO\\|XXX\\)\\)[[:space:]]+\\(.+\\)\\|\\(?:^\\|[[:blank:]]+\\)\\(\\(?:DEBUG\\|FIXME\\|GOTCHA\\|STUB\\|TODO\\|XXX\\)\\)\\(?:\\(?:[([][^])]+[])]\\)?:\\)\\(?:[[:blank:]]+\\(.+\\)\\)?\\)")
  magit-todos--scan-with-rg(:callback magit-todos--insert-items :magit-status-buffer #<buffer magit: emacs> :directory "/home/cj/.config/emacs/" :depth nil)
  #f(compiled-function () #<bytecode -0x4ba71d4c1062852>)()
  magit-todos--insert-todos()
  magit-run-section-hook(magit-status-sections-hook)
  (catch 'cancel-section (magit-run-section-hook 'magit-status-sections-hook) (magit-insert-section--finish section111))
```

</details>

**Comment by alphapapa** [comment]
I just tested using a clean Emacs configuration with the latest version of all dependencies and it worked fine.  It appears that you need to reinstall or recompile other packages in your Emacs configuration.  Since the `rxt-` symbols are part of the `pcre2el` package, I'd suggest starting with that one.

**Comment by JimMoen** [comment]
Have tried `M-x package-reinstall magit-todos RET` and `M-x package-reinstall pcre2el RET` but not work.
I'll trying rm all eln cache files and reinstall.
Thanks a lot for your comment.

**Comment by alphapapa** [comment]
Emacs 30 is not yet tagged nor released, but you are running Emacs "31.0.50."  I cannot support unreleased versions of Emacs.  As I said, I have verified that this package works correctly with the latest version of all dependencies, and that on Emacs 29.  If you insist on running snapshot builds of `master` years before that tree will be tagged and released, you are on your own.

**Comment by furkanusta** [comment]
@JimMoen 
I had this problem in commit 4e250154ec4 of Emacs and I've done a pull recently at 3ad4623bec4 the error is gone

Not sure which commit your package uses but you might try updating it
